src/main/java/seedu/address/logic/commands/UnmarkAttendanceByStudentCommand.java

```
@@ -0,0 +1,111 @@
+            + PREFIX_TUTORIAL + " Math";
+
+    public static final String MESSAGE_UNMARK_ATTENDANCE_STUDENT_SUCCESS =
+            "Unmarked attendance of %1$s student for %2$s tutorial on %3$s";
+
+    public static final String MESSAGE_INVALID_TUTORIAL_FOR_STUDENT =
+            "The student does not take %1$s tutorial";
+
+    private final Index targetIndex;
+    private final Attendance attendance;
+    private final Tutorial tutorial;
+
+    /**
+     * @param targetIndex Index of the person in the filtered person list to mark
+     * @param attendance Attendance of the person specified by index
+     * @param tutorial Tutorial the student attended
+     */
+    public UnmarkAttendanceByStudentCommand(Index targetIndex, Attendance attendance, Tutorial tutorial) {
+        requireAllNonNull(targetIndex, attendance, tutorial);
+        this.targetIndex = targetIndex;
+        this.attendance = attendance;
+        this.tutorial = tutorial;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person studentToUnmarkAttendance = lastShownList.get(targetIndex.getZeroBased());
+        Participation currentParticipation = model.getParticipationList().stream()
+                .filter(participation -> participation.getStudent().equals(studentToUnmarkAttendance)
+                        && participation.getTutorial().equals(this.tutorial)).findFirst()
+                .orElseThrow(() -> new CommandException(String.format(MESSAGE_INVALID_TUTORIAL_FOR_STUDENT, tutorial)));
+
+        List<Attendance> updatedAttendance = new ArrayList<>(currentParticipation.getAttendanceList());
+        updatedAttendance.add(attendance);
```

Here should be removing attendance?

---

Yea I think we should remove the attendance here and throw an exception if the attendance for that date is not found

---

Thanks for the catch! Fixed now.